### PR TITLE
feat: Cache Manager

### PR DESCRIPTION
**Pull request by bmgalego**
This PR introduces a new cache manager to the agent runtime, centralizing most of the caching logic into a dedicated manager. Currently, this implementation only covers the following caches, twitter cache, discord cache, solana cache, browser and video cache and does not include files like video, audio and other attachments.
The cache manager provides a straightforward API (still a work in progress) that uses a cache adapter. For now using a filesystem adapter to test it awaiting for feedback to start writing the schemas and db adapters.
The caching api uses a key and the agentId to index the entries allowing for multi agent caching.
We could also include in the api an extra argument for expiration.

# Relates to:

#243
#258
This pr should fix this issue since the multi agent twitter caching no longer conflicts, which was likely the root cause of the bug.

# Risks
medium – this affects multiple packages.

# Background

## What kind of change is this?

Feature: Introduces a new caching manager.
Improvements: Refactors and centralizes caching logic for multiple packages.
Bug Fix: Resolves conflicts in multi-agent Twitter caching

# Documentation changes needed?
My changes require a change to the project documentation.

# Testing

## Where should a reviewer start?
packages/core/src/cache.ts – Review the API implementation.
packages/agent/src/index.ts – Check how the cache manager is instantiated and integrated.
packages/client-twitter/src/base.ts – Verify the usage of the cache manager API in the Twitter client.

### CODE REVIEW

**Comment by ponderingdemocritus** [comment]
good idea and correct direction -  lets make this work then merge up

**Comment by bmgalego** [comment]
Added some basic tests and implemented the expiration api.
I think initialy we could go without the database adapters, starting with the filesystem should be easier to migrate too. I am building a script to help the migration.

File: packages/agent/src/index.ts [hunk 1]
```diff
@@ -159,13 +171,37 @@ export function getTokenForProvider(
...
+        plugins: [],
+        providers: [],
+        actions: [],
+        services: [],
+        managers: [],
+        cacheManager: cache,
+    });
+}
+
+function initializeDatabase(dataDir: string) {
```
Comment (ponderingdemocritus) [inline packages/agent/src/index.ts:197]: this isn't needed anymore

**Review by ponderingdemocritus — COMMENTED** [review]
epic PR - some conflicts. Could adjust all logging to use elizaLogger.

I don't have many other comments on implementation. Lets test in the wild across all services and clients to make sure working as expected

**Comment by monilpat** [comment]
Thanks so much for working on this! Looks like there are some typing issues that I am fixing as a part of merging in my latest changes just wanted to give a heads up

**Comment by monilpat** [comment]
I think to prevent issues like this we need to 1) run npm run build inside of each tsup.config.ts where we changed the types so that packages/adapter-postgres/dist/index.d.ts updates properly thanks again :)

**Pull request merged**